package strategy_protocol;

import "Common.proto";

message WorldStatusUpdate {
	message CommonProperties {
		optional string id = 1;
		optional Position position = 2;
	}

	message Base {
		optional CommonProperties common = 1;
		optional int32 health = 2;
	}

	message Mine {
		optional CommonProperties common = 1;
		optional int32 capacity = 2;
	}

	message Obstacle {
		optional CommonProperties common = 1;
	}

	message Worker {
		optional CommonProperties common = 1;
		optional int32 resource = 2;
		optional int32 health = 3;
	}

	message Soldier {
		optional CommonProperties common = 1;
		optional int32 health = 2;
	}

	message Boss {
		optional CommonProperties common = 1;
		optional int32 health = 2;
	}

	message Minion {
		optional CommonProperties common = 1;
		optional int32 health = 2;
	}

	repeated Base bases = 1;
	repeated Mine mines = 2;
	repeated Obstacle obstacles = 3;
	repeated Worker workers = 4;
	repeated Soldier soldiers = 5;
	repeated Boss bosses = 6;
	repeated Minion minions = 7;
	optional int32 resource = 8;
}

